invenio_cli/commands/install.py

```diff
@@ -21,14 +21,14 @@ def __init__(self, cli_config):
         """Constructor."""
         super(InstallCommands, self).__init__(cli_config)
 
-    def install_py_dependencies(self, pre):
+    def install_py_dependencies(self, pre, dev=False):
         """Install Python dependencies."""
-        steps = [
-            FunctionStep(
-                func=PackagesCommands.is_locked,
-                message="Checking if dependencies are locked."
-            )
-        ]
+        # If not locked, lock. Then install.
+        steps = []
+
+        if PackagesCommands.is_locked().status_code > 0:
+            steps.extend(PackagesCommands.lock(pre, dev))
+
         steps.extend(PackagesCommands.install_locked_dependencies(pre))
```

Comment:
This command should also take `dev` then if we introduce `dev` support, right?

Reply:
Changes added. Note that there is one `dev` and `developement` flags... I agree that is an issue that should be addressed, but will be done as part of: https://github.com/inveniosoftware/invenio-cli/issues/199

Reply:
Ah, my comment was for `PackagesCommands.install_locked_dependencies(pre)` i.e. it should be: `PackagesCommands.install_locked_dependencies(pre, dev)` and `install_locked_dependencies` should be changed otherwise the locked dev dependencies will not be installed despite being locked. 

I saw #199 more about naming and conventions. Adding the ability to install `--dev` dependencies is separate I think... There is disambiguation to be done between "development (symlink)" installs and "dev (dev dependencies)" for sure.